oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -184,6 +187,20 @@ public static GoogleCredentials fromStream(
             fileType, USER_FILE_TYPE, SERVICE_ACCOUNT_FILE_TYPE));
   }
 
+  /**
+   * Creates a credential with quota project from environment if present
+   *
+   * @param environmentProvider a provider used to get environment values
+   * @return credential with quota project from envoronment
+   */
+  public GoogleCredentials createWithQuotaProject(String quotaProject) {
+    if (quotaProject == null) {
```

Comment:
Looking at the test, this behavior seems a bit error prone. Calling this will `null` with neither clear the entry or give me a new instance.

Either way, this behavior should be described in the javadoc.

Reply:
Updated the behavior to set null for the quota project as well. This was lingering behavior from an itr where env var was read here. Thanks for catching